app/dashboard/utils.py

```diff
@@ -998,3 +999,33 @@ def list_urls(lis, acc=None):
 
 def get_custom_avatars(profile):
     return CustomAvatar.objects.filter(profile=profile).order_by('-id')
+
+
+def trim_null_address(address):
+    if address == '0x0000000000000000000000000000000000000000':
+        return '0x0'
+    else:
+        return address
+
+
+def get_token_recipient_senders(recipient_address, token_addres):
+    w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
+    contract = w3.eth.contract(
+        address=token_address,
+        abi=erc20_abi,
+    )
+
+    balance = contract.functions.balanceOf(recipient_address).call()
+
+    if balance == 0:
+        return []
+
+    transfers = contract.events.Transfer.getLogs(
```

Comment:
AttributeError: type object 'Transfer' has no attribute 'getLogs'

Reply:
what version of web3py are you running? im running 4.6.0, which is the version in the requirements/base.txt

Reply:
Oops, I wrote this using the latest web3. Ill get this fixed later today.